HarryPotter/Screens/House/ViewController/HouseViewController.swift

```diff
@@ -91,17 +86,29 @@ extension HouseViewController {
         return viewModel.characters.count
     }
     
+    //TODO: Complete the function
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: cellId, for: indexPath) as! HouseMemberCell
         let currentCharacter = viewModel.characters[indexPath.row]
         cell.myLabel.text = currentCharacter.name
         cell.backgroundColor = viewModel.backgroundColor
         cell.selectionStyle = .none
-        cell.markedFavourite = {
-            print(currentCharacter.name)
-            if let encoded = try? JSONEncoder().encode(currentCharacter) {
-                UserDefaults.standard.set(encoded, forKey: self.viewModel.userKey)
+    
+        let person = viewModel.characters[indexPath.row]
+        cell.setPersonLabel(text: person.name.capitalized)
+        let isAlreadyFavourite = dependencies.dataController.isSaved(person: person)
+        cell.setButtonState(isSelected: isAlreadyFavourite)
+        cell.makedFavourite = { [weak self] shouldDelete in
+            do {
+                if shouldDelete {
+                    try self?.dependencies.dataController.delete(person: person)
```

Comment:
Could you try to remove `ApplicationDependencies` from ViewControllers? So here, you would ask `try self?.viewModel.delete(person: person)` and the viewModel could have a `func delete(person: Person)` that would ask `dependencies.DataController.delete(person: person)`.

The goal is move all the important stuff to ViewModel ;)